### PR TITLE
Style guide added

### PR DESCRIPTION
**Pull request by sushmangupta**
Please review the style guide for the tech doc created.

### CODE REVIEW

**Review by elkmod — CHANGES_REQUESTED** [review]

File: resources/guidelines/documentation-guidelines/04-fonts-and-formats/02-code.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
+# Fonts and Format for Codes
+
+## Code Format
+
+* To mark a block of code such as a lengthy command or a code sample, use a code fence (```) in Markdown.
```
[the remark this answers is not in the document]
Comment (sushmangupta) [reply resources/guidelines/documentation-guidelines/04-fonts-and-formats/02-code.md:5]: okay

**Comment by sushmangupta** [comment]
@elkmod, this is the final draft of the style guide. All the review comments from you and Andrea are incorporated. Some additional information is also newly added. The structure is also lightly changed. Please have a look at it.

File: resources/guidelines/documentation-guidelines/01-general.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+It is important to consider the background of the potential audience reading your writing. This helps you to adapt your writing to meet their needs and interests.
+
+| Who is the audience? | What are their roles? |
+| :--- | :--- |
+| Fullstack developer | <ul><li>Plugin development</li><li>Templates</li><li>Routes/ Controllers</li></ul>|
+| Frontend developer | <ul><li>Admin</li><li>Themes</li><li>PWA</li></ul>  |
+| Backend developer | <ul><li>DI / Service architecture</li><li>Message queues</li><li>DAL</li><li>Action event system</li><li>ElasticSearch</li></ul> |
+| API developer  | <ul><li>How to consume the API</li><li>Create a product / category import</li><li>How to extend the API</li><li>API paradigm</li><li>Proper API references</li><li>Request collection</li></ul> |
+| DevOps  | <ul><li>Hosting setup</li><li>Deployment</li><li>Performance tests</li></ul> |
+| Project/Solution architect | <ul><li>Hosting</li><li>Architecture</li><li>Modules / Extension system</li><li>Paradigms / patterns</li><li>Commonalities app system / Plugin system</li></ul> |
```
Comment (elkmod) [inline resources/guidelines/documentation-guidelines/01-general.md:18]: Inconsistent placement of spaces around `/` separator - also with `Product owner/ manager` in line 20.

File: resources/guidelines/documentation-guidelines/01-general.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+## List of docs maintained by us
+
+* [PWA](https://shopware-pwa-docs.vuestorefront.io/)
+
+* [Developer Guide](https://developer.shopware.com/docs/)
+
+* [API Reference Guide](https://shopware.stoplight.io/)
+
+* [Admin Extension SDK](https://shopware.github.io/admin-extension-sdk/)
```
Comment (elkmod) [inline resources/guidelines/documentation-guidelines/01-general.md:31]:
```suggestion
* [Developer Guide](https://developer.shopware.com/docs/)
* [API Reference Guide](https://shopware.stoplight.io/)
* [PWA](https://shopware-pwa-docs.vuestorefront.io/)
* [Admin Extension SDK](https://shopware.github.io/admin-extension-sdk/)
```

File: resources/guidelines/documentation-guidelines/01-general.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+## Word list
+
+Choose ecommerce and technical terms from the pre-defined list of terminologies:
+
+* [Shopware terminologies](https://shopware.atlassian.net/wiki/spaces/pr/pages/19249037615/Shopware+terminology)
+
```
Comment (elkmod) [inline resources/guidelines/documentation-guidelines/01-general.md:38]: Lists don't need an extra space, they are displayed the same in the frontend.

File: resources/guidelines/documentation-guidelines/01-general.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+## Use of third-party sources
+
+Third-party sources include websites, books, blogs, videos, images, and more. Ensure to reference these external sources in the documentation only if they are trustworthy. Avoid copying any content directly from other sources like websites, encyclopedias, and Wikipedia.
+
+## Markdown rules
+
+Adhere to the [Markdown cheat sheet](https://www.markdownguide.org/cheat-sheet/) while creating the document.
+
+Refer to [gitbook syntax](https://gitbook.gitbook.io/git-sync/) for features like hint block, emoji, API blocks, etc.
```
Comment (elkmod) [inline resources/guidelines/documentation-guidelines/01-general.md:49]:
```suggestion
Refer to [GitBook syntax](https://gitbook.gitbook.io/git-sync/) for features like hint block, emoji, API blocks, etc.
```

**Review by elkmod — CHANGES_REQUESTED** [review]

File: resources/guidelines/documentation-guidelines/01-general.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+* [Developer Guide](https://developer.shopware.com/docs/)
+
+* [API Reference Guide](https://shopware.stoplight.io/)
+
+* [Admin Extension SDK](https://shopware.github.io/admin-extension-sdk/)
+
+## Word list
+
+Choose ecommerce and technical terms from the pre-defined list of terminologies:
```
Comment (elkmod) [inline resources/guidelines/documentation-guidelines/01-general.md:35]: We can add a hint here, indicating that these are internal resources only visible to Shopware employees.

File: resources/guidelines/documentation-guidelines/04-fonts-and-formats/02-code.md [hunk 7]
```diff
@@ -0,0 +1,165 @@
+# Fonts and Format for Code
+
+Fonts and formats for inline code, code blocks, non-code items, API referece, classes and methods are detailed below:
+
+## Inline code
+
+* Inline code is a short snippet of code. Use backticks (`) for single-line code/ inline code.
```
Comment (elkmod) [inline resources/guidelines/documentation-guidelines/04-fonts-and-formats/02-code.md:7]: I think this has to be escaped properly (see highlighting in GitHub)

File: resources/guidelines/documentation-guidelines/04-fonts-and-formats/02-code.md [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+
+* Don't show the current directory path before the prompt, even if part of the instruction includes changing directories.
+
+### Placeholders
+
+* In a code output, explain any placeholder that appears in the sample output the first time.
+
+* Mention the placeholders in complete capital and italicized code font.
+
+* In markdown, wrap inline placeholders in backticks (\`) and asterisk (*).
```
Comment (elkmod) [inline resources/guidelines/documentation-guidelines/04-fonts-and-formats/02-code.md:65]: Also here

File: resources/guidelines/documentation-guidelines/04-fonts-and-formats/02-code.md [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+
+* Method names should be followed by a pair of parentheses `()`.
+
+* You may also cross-link parameters, classes, and methods.
+
+## Deprecations
+
+When something is deprecated, tell the user what to use as a replacement or what to do to make their code work. For example,
+
+{% hint style="" %}
```
Comment (elkmod) [inline resources/guidelines/documentation-guidelines/04-fonts-and-formats/02-code.md:159]:
```suggestion
{% hint style="warning" %}
```

File: resources/guidelines/documentation-guidelines/04-fonts-and-formats/02-code.md [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+
+* Transaction
+
+{% endhint %}
+
+* Don't use tabs to indent text within a code block; use two spaces.
+
+* Use three dots (...) on a separate line to indicate that more lines of output are omitted from the sample output.
+
+* Refer to [Syntax & Guidelines](https://handbook.shopware.com/Product/Guides/Development/DeveloperDocumentation) for more examples.
```
Comment (elkmod) [inline resources/guidelines/documentation-guidelines/04-fonts-and-formats/02-code.md:119]:
```suggestion
* Refer to [Syntax & Guidelines](https://handbook.shopware.com/Product/Guides/Development/DeveloperDocumentation#syntax--guidelines) for more examples.
```

File: resources/guidelines/documentation-guidelines/06-doc-process.md [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+
+All contents are based on Shopware Major versions, such as 6.3, 6.4, 6.5, etc. The current version is reflected by our GitHub repositories' `master` branch, whereas each older version has its respective separate branch.
+
+If a documented feature or functionality is introduced within major versions (and also in cases where you think it is applicable), please include a hint showing the version constraints as below:
+
+{% hint style="info" %}
+This functionality is available starting with Shopware 6.4.3.0.
+{% endhint %}
+
+Your Contribution is our Pride!
```
Comment (elkmod) [inline resources/guidelines/documentation-guidelines/06-doc-process.md:65]: Capitalization of nouns?

**Review by elkmod — CHANGES_REQUESTED** [review]

File: resources/guidelines/documentation-guidelines/06-doc-process.md (hunk 11 above)
Comment (sushmangupta) [reply resources/guidelines/documentation-guidelines/06-doc-process.md]: Ohhh.. now that is ironic after all the standards are set :P . However, my intention was to make it sound prominent. I have removed it and made it bold rather.

**Review by elkmod — APPROVED** [review]

**Pull request merged**